game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/general/FiringGroupSplitterGeneral.java

```diff
@@ -0,0 +1,135 @@
+package games.strategy.triplea.delegate.battle.steps.fire.general;
+
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.Side.OFFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.UnitBattleFilter.ACTIVE;
+import static games.strategy.triplea.delegate.battle.BattleState.UnitBattleFilter.ALIVE;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.AIR_FIRE_NON_SUBS;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.steps.fire.FiringGroup;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import lombok.Value;
+import org.triplea.java.PredicateBuilder;
+import org.triplea.java.collections.CollectionUtils;
+
+/**
+ * Create firing groups with units that match {@link #firingUnitPredicate}
+ *
+ * <p>The firing groups are separated by canNotTarget, canNotBeTargetedBy, and isSuicideOnHit
+ */
+@Value(staticConstructor = "of")
+public class FiringGroupSplitterGeneral implements Function<BattleState, List<FiringGroup>> {
+
+  BattleState.Side side;
+
+  Predicate<Unit> firingUnitPredicate;
+
+  /** Name displayed in the Battle UI */
+  String groupName;
+
+  @Override
+  public List<FiringGroup> apply(final BattleState battleState) {
```

Comment:
The original logic that this is based off of is in `MustFightBattle#findTargetGroupsAndFire` (https://github.com/triplea-game/triplea/blob/e160d341f6c60ecf5297aad9ab26ac5a524dc2dd/game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java#L1638) and `MustFightBattle#fire` (https://github.com/triplea-game/triplea/blob/e160d341f6c60ecf5297aad9ab26ac5a524dc2dd/game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java#L1497)